sop-core/src/Data/SOP/NS.hs

```diff
@@ -201,8 +278,7 @@ shiftEjection (Fn f) = Fn $ (\ns -> case ns of Z _ -> Comp Nothing; S s -> f (K
 -- @since 0.2.2.0
 --
 unZ :: NS f '[x] -> f x
-unZ (Z x) = x
-unZ (S x) = case x of {}
+unZ (NS _ x) = unsafeCoerce x
 
 -- | Obtain the index from an n-ary sum.
 --
```

Comment:
Commenting here for the `index_NS` definition below, but this PR should also have:

```haskell
index_NS :: forall f xs . NS f xs -> Int
index_NS (NS i _) = i
```